module/zfs/vdev.c

```diff
@@ -795,6 +795,7 @@ vdev_alloc(spa_t *spa, vdev_t **vdp, nvlist_t *nv, vdev_t *parent, uint_t id,
 	vd = vdev_alloc_common(spa, id, guid, ops);
 	vd->vdev_tsd = tsd;
 	vd->vdev_islog = islog;
+	vd->vdev_isspecial = (alloc_bias == VDEV_BIAS_SPECIAL) ? 1 : 0;
```

Comment:
Aside of VDEV_BIAS_SPECIAL there is also VDEV_BIAS_DEDUP, that is not less special from the point of spare replace.

Reply:
vdev already has vdev_alloc_bias field, that you should be able to just check.